api/cpp/esp-idf/slint/README.md

```diff
@@ -11,80 +11,139 @@ It has been tested on ESP32-S3 devices.
 
 ![Screenshot](https://user-images.githubusercontent.com/959326/260754861-e2130cce-9d2b-4925-9536-88293818ac3e.jpeg)
 
-## Prerequisites
+## Getting Started
 
-In order to compile Slint, you need a working [Rust toolchain for your device](https://esp-rs.github.io/book/installation/index.html).
-Follow the instructions from <https://github.com/esp-rs/espup#installation> to set it up.
+### Prerequisites
 
-You can then either set the esp toolchain as the default toolchain with `rustup` or create a `rust-toolchain.toml` file in the root directory of your project with the following contents:
+* Install the [Espressif IoT Development Framework](https://docs.espressif.com/projects/esp-idf/en/latest/esp32/index.html) and open a terminal or command prompt with the environment set up.
+On Windows, follow the [Using the Command Prompt](https://docs.espressif.com/projects/esp-idf/en/latest/esp32/get-started/windows-setup.html#using-the-command-prompt) instructions, on macOS and Linux, follow the
+[Set up the Environment Variables](https://docs.espressif.com/projects/esp-idf/en/latest/esp32/get-started/linux-macos-setup.html#step-4-set-up-the-environment-variables) instructions.
+* Make sure that you have [Rust](https://esp-rs.github.io/book/installation/rust.html) installed.
```

Comment:
I think this could get a note that it needs to be the esp toolchain and not just "rust"

Reply:
I think the bullet point right after that covers that, no?

Reply:
Then I don't think this is necessary.  Having only the esp toolchain should be enough

Reply:
Can you elaborate what you mean with that?

How do you use espup without a regular installation of Rust?

When I try to do this on a Linux system (download espup binary on a system without any Rust installation), I get this:

```bash
$ ./espup install
[2023-08-24T06:44:45Z INFO ]  Installing the Espressif Rust ecosystem
[2023-08-24T06:44:45Z INFO ]  Querying GitHub API: 'https://api.github.com/repos/esp-rs/rust-build/releases/latest'
[2023-08-24T06:44:45Z INFO ]  Querying GitHub API: 'https://api.github.com/repos/esp-rs/rust-build/releases'
[2023-08-24T06:44:46Z INFO ]  Checking Rust installation
Error: espup::toolchain::rust::missing_rust

  x  Rust is not installed. Please, install Rust via rustup: https://rustup.rs/
```

This step, making sure that Rust is installed via rustup, is what the link in the bullet point we're commenting on covers.

Is there another espup installation step that I'm missing and that we should recommend instead, that doesn't require Rust?

Reply:
ah then you need rust, sorry for the noise

Reply:
No worries :)